proxy/src/source.rs

```diff
@@ -171,50 +172,49 @@ where
         id
     }
 
-    fn get_all_projects(&self) -> Vec<Project> {
-        // TODO(xla): Return proper error.
-        self.registry_client
+    fn get_all_projects(&self) -> Result<Vec<Project>, RegistryError> {
+         Ok(self
+            .registry_client
             .list_projects()
-            .wait()
-            .expect("osc client list projects failed")
+            .wait()? // bubble error up if registry client throws anything
             .into_iter()
             .take(10)
             .flat_map(|id| {
-                let maybe_project = self
-                    .registry_client
-                    .get_project(id)
-                    .wait()
-                    .expect("get_project failed");
-
-                match maybe_project {
-                    Some(project) => Some(self.enrich_members(Project::from(project))),
-                    None => None,
+                match self.get_project(id.into()) {
```

Comment:
This can be `let maybe_project = self.get_project(id.into())?`.

Any time you do `match` followed by `Err(err) => Err(err)` you should be able to convert that to using a `?` :)